software/o_c_REV/APP_ENVGEN.ino

```diff
@@ -192,9 +198,19 @@ public:
 
     OC::DigitalInput trigger_input = get_trigger_input();
     uint8_t gate_state = 0;
-    if (triggers & DIGITAL_INPUT_MASK(trigger_input))
-      gate_state |= peaks::CONTROL_GATE_RISING;
 
+    bool triggered = triggered_;
+    if (triggers & DIGITAL_INPUT_MASK(trigger_input)) {
+      sinceTrigger_ = 0;
+      triggered=true;
+    }
+    uint16_t trigger_delay = get_trigger_delay();
+    if ((!trigger_delay && triggered) || (triggered && (sinceTrigger_ >= trigger_delay))) {
```

Comment:
> On 4 Apr 2016, at 11:38 PM, Patrick Dowling <notifications@github.com> wrote:
> 
> In software/o_c_REV/APP_ENVGEN.ino <https://github.com/patrickdowling/O_C/pull/18#discussion_r58376065>:
> 
> >  
> > +    bool triggered = triggered_;
> > +    if (triggers & DIGITAL_INPUT_MASK(trigger_input)) {
> > +      sinceTrigger_ = 0;
> > +      triggered=true;
> > +    }
> > +    uint16_t trigger_delay = get_trigger_delay();
> > +    if ((!trigger_delay && triggered) || (triggered && (sinceTrigger_ >= trigger_delay))) {
> A lot of Arduino libraries/code isn't safe to use in ISRs so I tend to stay away from it. If that's from Paul it might be ok.
> 
> 

Dunno. It passes the “seems to work” functional test, but can be re-written just to use a counter if you think that’s safer.

Tim C